service/controller/resource/capzcrs/create.go

```diff
@@ -290,6 +294,109 @@ func (r *Resource) updateCRs(ctx context.Context, desiredCRs []runtime.Object) e
 	return nil
 }
 
+func detectAzureClusterUpdate(orig, desired runtime.Object) (bool, error) {
+	o := orig.(*capzv1alpha3.AzureCluster)
+	d := desired.(*capzv1alpha3.AzureCluster)
+
+	if !cmp.Equal(d.GetLabels(), o.GetLabels()) {
+		return true, nil
+	}
+
+	if !cmp.Equal(d.GetAnnotations(), o.GetAnnotations()) {
+		return true, nil
+	}
+
+	if !cmp.Equal(d, o, cmpopts.IgnoreTypes(&metav1.ObjectMeta{}), cmpopts.IgnoreTypes(&metav1.TypeMeta{}, cmpopts.IgnoreTypes(&capzv1alpha3.Subnets{}))) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func mergeAzureCluster(orig, desired runtime.Object) (runtime.Object, error) {
```

Comment:
The func will always take subnets from original and set them to desired, right? There isn't an use case where subnets from both original and desired will be taken?

Reply:
Nope. More details above :slightly_smiling_face: